sky/compositor/layer.h

```diff
@@ -46,10 +46,15 @@ class Layer {
     paint_bounds_ = paint_bounds;
   }
 
+ protected:
+  virtual void Paint(PaintContext& context) = 0;
+
  private:
   ContainerLayer* parent_;
   SkRect paint_bounds_;
 
+  friend class ContainerLayer;
```

Comment:
Why do we need to friend ContainerLayer?  It's already a subclass.  Can't we just make whatever it needs to see protected?